### PR TITLE
fix(core): restore values on update if needed

### PR DESCRIPTION
**Pull request by stonebuzz**
restore values on update if needed

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | internal 21623

### CODE REVIEW

**Comment by trasher** [comment]
Tests are failing

**Comment by stonebuzz** [comment]
All fields are restored except for actors fields.

I don't think it can be done "easily".

Unless you have an idea ?

File: inc/commonglpi.class.php [hunk 1]
```diff
@@ -842,8 +842,6 @@ function showTabsContent($options = []) {
             unset($cleaned_options['stock_image']);
          }
          if ($this->getType() == 'Ticket') {
-            $this->input = $cleaned_options;
-            $this->saveInput();
```
Comment (cedric-anne) [inline inc/commonglpi.class.php:846]: This is a "hack", and removing it causes description field content emptyness when form is reloaded (when category is changed, for instance).

File: inc/ticket.class.php [hunk 2]
```diff
@@ -4184,6 +4184,15 @@ function showForm($ID, $options = []) {
...
       }
 
+      // Restore saved value and override $this->fields
+      if (count($saved)) {
+         foreach ($saved as $name => $value) {
+            if (isset($this->fields[$name])) {
+               $this->fields[$name] = $saved[$name];
+            }
+         }
+      }
```
Comment (cedric-anne) [inline inc/ticket.class.php:4194]: Maybe this piece of code, from line 4177 to 4194, should be mutualized with Change and Problem classe. See #8771 where I add to add save input restoration process on these classes.

File: inc/commonglpi.class.php (hunk 1 above)
Comment (stonebuzz) [reply inc/commonglpi.class.php:846]: Ok, I propose to use this code only when creating a ticket
done

File: inc/ticket.class.php (hunk 2 above)
Comment (stonebuzz) [reply inc/ticket.class.php:4194]: ok, so it is now possible to restore the values for changes and problems
done

File: inc/commonglpi.class.php (hunk 1 above)
Comment (cedric-anne) [reply inc/commonglpi.class.php]: `$this->input = $cleaned_options;` is used later to restore uploaded files (`$uploads['_filename'] = $this->input['_filename'];`). 

On update, as `$this->input` is not set, files uploaded before form reload are not restored and so images are stored as base64 in rich text.

I guess the solution would be to set `$this->input` on `restoreSavedValues` method.

**Review by cedric-anne — APPROVED** [review]

File: inc/commondbtm.class.php [hunk 3]
```diff
@@ -1037,6 +1037,28 @@ protected function restoreInput(Array $default = []) {
       return $default;
    }
 
+   /**
+    * Restore data saved in the session to $this->input
+    *
+    * @since 9.5.4
```
Comment (orthagh) [inline inc/commondbtm.class.php:1043]:
```suggestion
    * @since 9.5.5
```

**Review by orthagh — CHANGES_REQUESTED** [review]

**Review by trasher — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**